flexget/plugins/api_tvmaze.py

```diff
@@ -0,0 +1,402 @@
+from sqlalchemy import Column, Integer, DateTime, String, Unicode, ForeignKey, Numeric, PickleType, func, Table, or_, \
+    and_
+from sqlalchemy.orm import relation
+
+from flexget import db_schema, plugin
+from flexget.event import event
+from flexget.utils.database import with_session
+from flexget.utils.tools import split_title_year
+
+log = logging.getLogger('api_tvmaze')
+
+DB_VERSION = 0
+Base = db_schema.versioned_base('tvmaze', DB_VERSION)
+UPDATE_INTERVAL = 7  # Used for expiration, number is in days
+
+
+class TVMazeGenre(Base):
+    __tablename__ = 'tvmaze_genres'
+
+    id = Column(Integer, primary_key=True, autoincrement=True)
+    name = Column(Unicode, unique=True)
+
+
+genres_table = Table('tvmaze_series_genres', Base.metadata,
+                     Column('series_id', Integer, ForeignKey('tvmaze_series.tvmaze_id')),
+                     Column('genre_id', Integer, ForeignKey('tvmaze_genres.id')))
+
+
+class TVMazeLookup(Base):
+    __tablename__ = 'tvmaze_lookup'
+
+    id = Column(Integer, primary_key=True, autoincrement=True)
+    search_name = Column(Unicode, index=True, unique=True)
+    series_id = Column(Integer, ForeignKey('tvmaze_series.tvmaze_id'))
+    series = relation('TVMazeSeries', backref='search_strings')
+
+
+class TVMazeSeries(Base):
+    __tablename__ = 'tvmaze_series'
+
+    tvmaze_id = Column(Integer, primary_key=True)
+    status = Column(Unicode)
+    rating = Column(Numeric)
+    genres = relation(TVMazeGenre, secondary=genres_table)
+    weight = Column(Integer)
+    updated = Column(DateTime)  # last time show was updated at tvmaze
+    name = Column(Unicode)
+    language = Column(Unicode)
+    schedule = Column(PickleType)
+    url = Column(String)
+    original_image = Column(String)
+    medium_image = Column(String)
+    tvdb_id = Column(Integer)
+    tvrage_id = Column(Integer)
+    premiered = Column(DateTime)
+    year = Column(Integer)
+    summary = Column(Unicode)
+    webchannel = Column(String)
+    runtime = Column(Integer)
+    show_type = Column(String)
+    network = Column(Unicode)
+    episodes = relation('TVMazeEpisodes', order_by='TVMazeEpisodes.season_number', cascade='all, delete, delete-orphan',
+                        backref='series')
+    last_update = Column(DateTime)  # last time we updated the db for the show
+
+    def __init__(self, series, session):
+        self.tvmaze_id = series.maze_id
+        self.update(series, session)
+
+    def update(self, series, session):
+        self.status = series.status
+        self.rating = series.rating['average']
+        self.weight = series.weight
+        self.updated = datetime.fromtimestamp(series.updated)
+        self.name = series.name
+        self.language = series.language
+        self.schedule = series.schedule
+        self.url = series.url
+        try:
+            self.original_image = series.image.get('original')
+        except AttributeError:
```

Comment:
This is a little bit non-obvious what it's catching. I think
`self.original_image = series.image and series.image.get('original')`
is better, or maybe being even more explicit is better
```
if series.image:
    self.original_image = series.image.get('original')
else:
    self.original_image = None
```